implementations/rust/ockam/ockam_api/src/nodes/service/http.rs

```diff
@@ -0,0 +1,119 @@
+use std::convert::Infallible;
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use http_body_util::combinators::BoxBody;
+use http_body_util::{BodyExt, Empty, Full};
+use hyper::body::Bytes;
+use hyper::server::conn::http1;
+use hyper::service::service_fn;
+use hyper::{Method, Request, Response};
+use hyper_util::rt::TokioIo;
+use ockam_core::AsyncTryClone;
+use ockam_node::Context;
+use serde::Serialize;
+use tokio::net::TcpListener;
+use tokio::task::JoinHandle;
+
+use crate::nodes::models::node::NodeResources;
+use crate::nodes::BackgroundNodeClient;
+use crate::{HttpError, Result};
+
+pub struct HttpServer {
```

Comment:
It would be good to leave a bit of documentation to explain:

 - Why do we need a HttpServer?
 - Is it supposed to support anything more than what it does now? (all the functions of the node manager API?)
 - Are there some constraints on its configuration (I'm thinking about the port. How do we make sure that there's no conflict?)

Reply:
> Are there some constraints on its configuration (I'm thinking about the port. How do we make sure that there's no conflict?)

About this, I have some open questions about the UX of the `node create` command, and I wasn't confident enough to make a decision, so I left the port to be assigned automatically. I'll work on this on a separate PR when we decide how to approach this. Some of the questions:

- Should the HTTP server be optional? (i.e. adding a flag like `--disable-http-server`?)
- How do we express this feature in the `node create` command? An argument to set the port `--http-server-port`? Default to `0` to automatically assign a free port? Will the user ever care about the HTTP server port?

Reply:
> Should the HTTP server be optional? 

My gut feeling is that:

 - It should be optional,`off` by default, with a `enable-http-access` option to give access to the `NodeManager` services via `HTTP` instead of the Ockam protocol. That option becomes then relevant is some production deployment scenarios when HTTP observability is required.
 
 - The port should be `0` by default and be assignable for k8s-like deployments where we can reference the port in the YAML deployment file so it's used as a probe.

Reply:
That sounds good to me. I would name the arguments `--enable-http-server` and `--http-server-port` perhaps?